packages/adblocker-benchmarks/run.js

```diff
@@ -141,24 +155,47 @@ async function debug(moduleId, rawLists) {
 
   for (let index = 0; index < requests.length; index += 1) {
     const { url, frameUrl, cpt } = requests[index];
+    const info = { index, url, frameUrl, cpt };
 
     if (!isSupportedUrl(url) || !isSupportedUrl(frameUrl)) {
+      if (!compare) {
+        output.push(info);
+      }
+
       continue;
     }
 
-    const match = engine.match({ type: WEBREQUEST_OPTIONS[cpt], frameUrl, url });
+    info.match = engine.match({ type: WEBREQUEST_OPTIONS[cpt], frameUrl, url });
 
-    let matchDebug = null;
+    info.matchDebug = info.match;
     if (engine.matchDebug) {
-      matchDebug = engine.matchDebug({ type: WEBREQUEST_OPTIONS[cpt], frameUrl, url });
+      info.matchDebug = engine.matchDebug({ type: WEBREQUEST_OPTIONS[cpt], frameUrl, url });
     }
 
-    output.push({ index, url, frameUrl, cpt, match, matchDebug });
+    if (compare !== null) {
+      if (info.match) {
```

Comment:
In compare mode, the basic idea is that if the value of `match` is not the same then we report other details about how it's not the same.

It's possible for an engine to give a different result as far as the exact filter matched is concerned. For example, given two filters `/foo^` and `/bar^` and the URL `https://example.com/foo/bar`, one engine might pick `/foo^` while another picks `/^bar`. But we don't care about such differences. We only care if one engine blocks and another engine allows.

Reply:
>We only care if one engine blocks and another engine allows.

Actually we do care in the case of redirection filters.

Maybe we'll add another option for showing mismatches at the filter level itself, but if this is the default output it's going to give lots of mismatches and it's not what we are interested in at the moment.